Section_2_Test_File.py

```diff
@@ -19,3 +19,5 @@ def test_is_isogram():
     print("All test cases passed")
 
 test_is_isogram()
+
+done
```

Comment:
8/12 => Your test file is not well structured you should have had something like this

```
import unittest
from Section_2 import isogram_checker
# importing the unittest module and the function we are testing

class TestingIsogramFunction(unittest.TestCase):

    def test_isogram_checker_true(self):
        expected = True
        result = isogram_checker("pie")
        self.assertEqual(expected, result)
    # checks that an isogram gets a true result using an example of "pie" as our input word

    def test_isogram_checker_false(self):
        expected = False
        result = isogram_checker("green")
        self.assertEqual(expected, result)
    # checks that a word that is not an isogram gets a false result using an example of "green"

    def test_isogram_checker_int(self):
        with self.assertRaises(TypeError):
            isogram_checker(4)
    # Tests than an invalid input (in this case if you use a int instead of a string) raises a type error

    def test_isogram_checker_edge_case_same_char(self):
        expected = False
        result = isogram_checker("eeeeeeeee")
        self.assertEqual(expected, result)
    # Tests for an edge case where instead of inputting a word, we just get a collection of the same letter - should return false

    def test_isogram_checker_edge_case_spaces(self):
        expected = True
        result = isogram_checker("pie mouth")
        self.assertEqual(expected, result)
    # Tests for an edge case where two words are used in the input
    # If i had more time I would expand the isogram function to split out words in a string and apply the function to each word in the string
```